pkg/virt-handler/vm.go

```diff
@@ -649,6 +649,18 @@ func (d *VirtualMachineController) updateVMIStatus(vmi *v1.VirtualMachineInstanc
 			vmi.Status.MigrationState.AbortStatus = v1.MigrationAbortStatus(migrationMetadata.AbortStatus)
 			vmi.Status.MigrationState.Completed = migrationMetadata.Completed
 			vmi.Status.MigrationState.Failed = migrationMetadata.Failed
+
+			// Add sample to migration histogram if migration was completed
+			if migrationMetadata.Completed {
+				timeSpent := migrationMetadata.StartTimestamp.Time.Sub(migrationMetadata.EndTimestamp.Time)
+				if migrationMetadata.Failed {
+					// migrationMetrics would come from pkg/monitoring/migrations/prometheus
+					// Not sure where to declare it though
+					migrationMetrics.observeMigrationDuration(float64(timeSpent/time.Second), vmi.Status.MigrationState.SourceNode, vmi.Status.MigrationState.TargetNode, "failed")
+				} else {
+					migrationMetrics.observeMigrationDuration(float64(timeSpent/time.Second), vmi.Status.MigrationState.SourceNode, vmi.Status.MigrationState.TargetNode, "successful")
+				}
+			}
```

Comment:
This won't compile, of course, because `migrationMetrics` wasn't declared anywhere.

I _think_ that it must be _declared at/added to_ the virt-controller component since I don't want each virt-handler to expose it with different values, but I'm not sure how to do that :grimacing: 

@rmohr @xpivarc @vladikr could you take a look at it when you have some time?